KenticoOnboardingApplication/src/KenticoOnboardingApplication.Services/Services/UpdateItemService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading.Tasks;
+using KenticoOnboardingApplication.Contracts.Helpers;
+using KenticoOnboardingApplication.Contracts.Models;
+using KenticoOnboardingApplication.Contracts.Repositories;
+using KenticoOnboardingApplication.Contracts.Services;
+using KenticoOnboardingApplication.Contracts.Services.Wrappers;
+
+namespace KenticoOnboardingApplication.Services.Services
+{
+    internal class UpdateItemService : IUpdateItemService
+    {
+        private readonly IListRepository _repository;
+        private readonly ITimeManager _timeManager;
+        private readonly IGetItemService _getItemService;
+
+        public UpdateItemService(IListRepository repository, ITimeManager timeManager, IGetItemService getItemService)
+        {
+            _repository = repository;
+            _timeManager = timeManager;
+            _getItemService = getItemService;
+        }
+
+        public async Task<RetrievedItem<Item>> UpdateItemAsync(Item item)
+        {
+            var retrievedItem = await _getItemService.GetItemAsync(item.Id);
+            if (!retrievedItem.WasFound)
+            {
+                return retrievedItem;
+            }
+
+            var updatedItem = CreateUpdatedItem(item.Id, item.Text, retrievedItem.Item.CreationTime);
+            var result = await _repository.UpdateItemAsync(updatedItem);
+
+            return result == null
+                ? RetrievedItem<Item>.Empty
+                : new RetrievedItem<Item>(result);
+        }
+
+        private Item CreateUpdatedItem(Guid id, string text, DateTime creationTime)
```

Comment:
This method does not scale well (with the number of properties in an item). Imagine you add 5 new properties in the `Item` and you have this private method used at 10 places. You have to write the same-ish `.X, ` 50 times.

What about having just two parameters:
```suggestion
        private Item CreateUpdatedItem(RetrievedItem<Item> existingItem, Item newItem)
```
This way, you would only write 5 new lines of code in the aforementioned scenario…